### ISSUE TITLE
Printing Too Far to The Left

### ISSUE THREAD

**Issue by CKidd368** [ExLibrisGroup/spineomatic-cloudapp#110]
Hey, I'm printing spine labels and the issue I'm having is that 6/8 columns and rows print fine, but the last two columns and rows are shifted to the left too much. I adjusted the layout settings a few times but every adjustment I made would not fix the last two columns. Is there a way I can move the last two columns more to the right of the labels? There's 56 labels total, 8 rows and 7 columns. Thank you

**Comment by CKidd368** [comment]
I appreciate it

**Comment by mgobat** [comment]
@CKidd368, I can't replicate the problem. It looks like you are left aligning the call number in the template. Is there any other special formatting in the template? If so, you can get the source code of the template by clicking the "..." in the toolbar and then clicking the "<>" icon in the toolbar when you are editing the template. Then share the source code here.

My test results:

<img width="603" alt="ckidd" src="https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/45852490/a89449a5-6261-4b2a-b333-4836353e81d3">

Do you have a particular label stock you are using? Manufacturer, stock number, etc? Maybe that information could help me replicate the problem. Thanks.

**Comment by CKidd368** [comment]
Here are the labels we are using and the source code

![20230524_134226](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/23d11dc7-efcc-454f-8622-ee77ed7fab04)
![20230524_134232](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/ee8ae058-415d-49b0-8261-bf1bf87b9fa2)
![20230522_140816](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/1cfbca16-61ef-4915-90b9-3dd8b2eed351)

**Comment by mgobat** [comment]
@CKidd368, try these settings for the layout:

Top margin: 0
Left margin: 0
Page width: 8.5
Label width: 1.05
Label height: 1.56
Horizontal gap: 0
Vertical gap: 0
Labels per page: 56
Left padding: 0.25

This one seems so simple; not sure why it's an issue.

**Comment by CKidd368** [comment]
I tried the new layout but unfortunately, it didn't resolve this issue. I know it seems simple, I don't know what the issue is exactly. When I make changes to the layout it seems to shift over some of the columns for some reason as you can see if you compare this image to the one I posted earlier in this discussion 

![20230531_083743](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/297097c0-6eef-40ea-8367-c283f49f8c41)

**Comment by mgobat** [comment]
@CKidd368, looking at the template you provided, I would try these layout settings:

Measurement units: inches
Top margin: .25
Left margin: .21
Page width: 8.5
Label width: 1.01
Label height: 1.5
Horizontal gap: 0
Vertical gap: 0
Labels per page: 56
Left padding: 0

What manufacturer and model printer are you using?

Thanks.

**Comment by CKidd368** [comment]
Here's what it looks like with the layout you recommended:  
![20230626_113938](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/77d0da3f-15bf-43a7-8c40-c85ab5eb8517)
It's printing too far to the left and too far up

When I go to print and enable "print with gridlines", it looks like it will print perfectly but doesn't. : 
![20230626_114103](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/cc0c1608-89e4-43a7-b181-853897242142)

This is the closest I got to it printing correctly. Still needs to go a little down and to the left. 
![20230626_114139](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/ed835b37-f0ed-4d62-b1e6-51956881c5e5)
For that I used these settings:
Measurement units: inches
Top margin: 0
Left margin: 0
Page width: 8.5
Label width: 1
Label height: 1.5
Horizontal gap: .05
Vertical gap: 0.060
Labels per page: 56
Left padding: 0.25

**Comment by mgobat** [comment]
@CKidd368, which printer driver are you using? PCL5e, PCLXL, or Postscript Level 3? Are there any printer settings/properties with which you could experiment with left and top settings?

**Comment by CKidd368** [comment]
We are using the Dell B5460dn Laser Printer XL driver. I can change the paper size width and layout. I don't see any options to change the top and left settings specifically

**Comment by mgobat** [comment]
@CKidd368, I was able to install the printer driver, but everything continues to look fine. Of course, I don't have the label stock you are using, nor the actual printer to test.

What's most interesting to me is that in the template you included earlier, the top margin is clearly .25 inches, and the left margin is .21 inches. Yet, you have best results with both of those set to 0. Do small increases in the top/left margins help at all or does it immediately throw the top/left beyond what you need?

**Comment by mgobat** [comment]
@CKidd368, do you have the "Enable marginless templates and layouts" switch on or off when you print? It should be off.

**Comment by mgobat** [comment]
@CKidd368, when I print labels using the Dell B5460dn printer driver, I see a "More settings" option in the print preview (in Chrome, Firefox and Edge browsers). If you see the same, do you have the correct "Paper size" selected and "None" selected for the "Margins"?

**Comment by CKidd368** [comment]
I'll have to get back to you with the results on Monday. (I'm trying to help our librarian with this issue and I don't have access to SpineOMatic on my computer.) I'll make sure to check if the correct paper size is selected and see if "enable marginless templates and layouts" is enabled. Thank you @mgobat

**Comment by CKidd368** [comment]
Hey, so I kept the exact layout: 
Measurement units: inches
Top margin: 0
Left margin: 0
Page width: 8.5
Label width: 1
Label height: 1.5
Horizontal gap: .05
Vertical gap: 0.060
Labels per page: 56
Left padding: 0.25

And enabled "Enable marginless templates and layouts." When I got to the print page, I clicked on "Margins" and selected "Custom margins" and set the top margin to .13, the bottom margin to .25, and the left and right margins to .25. The labels are now printing correctly. Thank you for all of your help!
![20230801_103126](https://github.com/ExLibrisGroup/spineomatic-cloudapp/assets/134000569/4fcc1b55-ffbb-4fc6-87f5-53b5f21976bb)
